cursor: Fix out-of-surface movement for unmanaged surfaces

jlindgren90 opened the pull request:
Fixes an issue when dragging an unmanaged XWayland surface (e.g. `kdiff3` toolbar) -- if the mouse moves outside the toolbar, the toolbar gets "stuck" and no longer follows the mouse.

The handling for unmanaged surfaces is pretty similar to what we do for layer-shell surfaces.

jlindgren90 commented:
/cc @Consolatis

Consolatis commented on src/cursor.c:
```diff
@@ -296,6 +296,10 @@ process_cursor_motion(struct server *server, uint32_t time)
+#if HAVE_XWAYLAND
+		} else if (server->seat.pressed.is_unmanaged) {
+			wlr_scene_node_coords(server->seat.pressed.node, &lx, &ly);
+#endif
```
I think this could be replaced by `if(node->parent == server->unmanaged_tree)` so we don't need the `is_managed` bool in `seat->pressed`.

Consolatis commented:
Regarding the other change with using `memset()` to clear out the struct:

I am not sure about the change, it simplifies things a bit but also makes it harder to grep / search for changes of those variables. If we want to keep using it I'd prefer some small function wrapper `lab_reset_pressed_button()` (or a better name) instead so its a bit more self-explaining when reading the code.

Changed the PR into draft for now, feel free to switch it back when ready to merge.

jlindgren90 commented:
Updated with suggestions from @Consolatis. I also split out `process_cursor_motion_out_of_surface()`, mainly to avoid confusion between different `node`/`surface`/`view` variables.

Consolatis commented:
> I also split out `process_cursor_motion_out_of_surface()`

Nice, that looks much better. I'll give the PR a test run but code LGTM so far.

Consolatis commented:
Tested with and without xwayland enabled (`meson configure -Dxwayland=disabled build_master`) and everything seems to work fine:
- out-of-surface for usual views
- out-of-surface for layershell clients
- out-of-surface for unmanaged x11 clients

Nice work! Merging.
